program/src/token.rs

```diff
@@ -41,24 +41,11 @@ impl PartialOrd for Rational {
 }
 
 impl Div for Rational {
-    type Output = Result<Rational>;
+    type Output = f64;
 
     fn div(self, rhs: Self) -> Self::Output {
-        if rhs.numerator == 0 {
-            panic!("Cannot divide by zero-valued `Rational`!");
-        }
-        let numerator = self
-            .numerator
-            .checked_mul(rhs.denominator)
-            .ok_or(ArithmeticError)?;
-        let denominator = self
-            .denominator
-            .checked_mul(rhs.numerator)
-            .ok_or(ArithmeticError)?;
-        Ok(Rational {
-            numerator,
-            denominator,
-        })
+        (self.numerator as f64 * rhs.denominator as f64)
+            / (self.denominator as f64 * rhs.numerator as f64)
```

Comment:
Can you add a test for this method as well? The previous version behaved unexpectedly, let’s make sure that this one does not. Also, please add a comment to explain why the output type is `f64` and not `Rational`. (Because the product of two rationals would need to be twice as wide, or lose precision, so we choose to make the loss of precision explicit in the types.)